R/rendering.R

```diff
@@ -482,31 +482,36 @@ plotCurve <- function(curveData, params, outFile = NA, ymin = NA, logDose = FALS
           function(x) 
             rgb(x[1], x[2], x[3], alpha=alpha))  
   }
-  if(is.null(plotColors) | length(plotColors) == 0) {
+
+  if (length(plotColors) > 1) {
+    requestedColors <- TRUE
+  } else if(is.null(plotColors) | length(plotColors) == 0) {
+    requestedColors <- FALSE
     plotColors <- "black"
     if(!is.null(racas::applicationSettings$server.curveRender.plotColors) && racas::applicationSettings$server.curveRender.usePlotColorsByDefault) {
       plotColors <- trimws(strsplit(racas::applicationSettings$server.curveRender.plotColors,",")[[1]])
     }
   }
 
-  if(is.na(mostRecentCurveColor)) {
+  if(is.na(mostRecentCurveColor && !requestedColors)) {
```

Comment:
This doesn't seem right.  is.na is wrapping both mostRecentCurveColor and !requestedColors (which is never NA in this code.